javascript/javascript1/week4/voiceassistant.js

```diff
@@ -0,0 +1,81 @@
+let toDo = ["gym", "football", "coding", "reading"];
+let myName;
+const helloCommand = "hello my name is";
+let num1;
+let num2;
+
+function getReply(command) {
+  if (command.indexOf(helloCommand) === 0) {
+    myName = command.slice(helloCommand.length + 1);
+    return "nice to meet you " + myName;
+  } else if (command === "what is your name") {
+    return "your name is " + myName;
+  } else if (command === "add fishing to my to do") {
+    toDo.push("fishing");
+    console.log(toDo);
+  } else if (command === "remove fishing from my to do") {
+    const toRemoveToDo = toDo.indexOf("fishing");
+    if (toRemoveToDo > -1) {
+      toDo.splice(toRemoveToDo, 4);
+    }
+    console.log(toDo);
+    return "Removed fishing from your todo";
+  } else if (command === "What is on my todo?") {
+    let toDoLength = toDo.length;
+    return;
+    "you have " + toDoLength + " todos - gym,football,coding and reading";
+  } else if (command === "What day is it today?") {
+    let d = new Date();
+    let weekday = new Array(7);
+    weekday[0] = "Sunday";
+    weekday[1] = "Monday";
+    weekday[2] = "Tuesday";
+    weekday[3] = "Wednesday";
+    weekday[4] = "Thursday";
+    weekday[5] = "Friday";
+    weekday[6] = "Saturday";
+    let month = new Array();
+    month[0] = "January";
+    month[1] = "February";
+    month[2] = "March";
+    month[3] = "April";
+    month[4] = "May";
+    month[5] = "June";
+    month[6] = "July";
+    month[7] = "August";
+    month[8] = "September";
+    month[9] = "October";
+    month[10] = "November";
+    month[11] = "December";
+    console.log(weekday[d.getDay()]);
+    console.log(
+      d.getDate() + " of " + month[d.getMonth()] + " " + d.getFullYear()
+    );
```

Comment:
There is a new method for `Date` at the new JavaScript version to converts a date to a string by using the current or specified locale which is `toLocaleDateString`, you can pass some options to specify the format you want. Example:
```js
  if (inputString === 'what day is it today?') {
    return new Date().toLocaleDateString('en-US', {
      day: 'numeric',
      year: 'numeric',
      month: 'long'
    });
  }
```